File: ABC/126/C.c
#include<stdio.h>
#include<math.h>

int main(){
  int N;
  double K;

  scanf("%d %lf", &N, &K);

  int i;

  double sum = 0;
  double lo;

  for(i = 1; i <= N; i++){
    lo = log2(K/i);
    if(lo < 0) lo = 0;
    if(lo != (int)lo) lo++;
    sum += 1./pow(2, (int)lo);
  }
  sum /= N;

  printf("%.10lf\n", sum);

  return 0;
}

File: ABC/126/C.py
import math

N, K = map(int, input().split())

prab = 0

for i in range(1, N+1):
    lo = math.log2(K/i)
    if lo < 0:
        lo = 0
    elif lo != int(lo):
        lo += 1

    prab += 1/pow(2., int(lo))

print(prab/N)
